docs/contributors/coding-guidelines.md

```diff
@@ -86,22 +86,32 @@ Example:
 import VisualEditor from '../visual-editor';
 ```
 
-### Experimental APIs
+### Experimental and Unstable APIs
 
-Exposed APIs that are still being tested, discussed and are subject to change should be prefixed with `__experimental`, until they are finalized. This is meant to discourage developers from relying on the API, because it might be removed or changed in the (near) future.
+Experimental and unstable APIs are temporary values exported from a module whose existence is either pending future revision or provides an immediate means to an end.
 
-Example:
+_To External Consumers:_
 
-```js
-export { __experimentalDoAction } from './api';
-```
+**There is no support commitment for experimental and unstable APIs.** They can and will be removed or changed without advance warning, including as part of a minor or patch release. As an external consumer, you should avoid these APIs.
+
+_To Project Contributors:_
 
-If an API must be exposed but is clearly not intended to be supported into the future, you may also use `__unstable` as a prefix to differentiate it from an experimental API. Unstable APIs should serve an immediate and temporary purpose. They should _never_ be used by plugin developers as they can be removed at any point without notice, and thus should be omitted from public-facing documentation. The inline code documentation should clearly caution their use.
+An experimental or unstable API is named as such to communicate instability of a function whose interface is not yet finalized. Aside from references within the code, these APIs should neither be documented nor mentioned in any CHANGELOG. They should effectively be considered to not exist from an external perspective. In most cases, they should only be exposed to satisfy requirements between packages maintained in this repository.
+
+An experimental or unstable function or object should be prefixed respectively using `__experimental` or `__unstable`.
 
 ```js
-export { __unstableDoAction } from './api';
+export { __experimentalDoExcitingExperimentalAction } from './api';
+export { __unstableDoTerribleAwfulAction } from './api';
```

Comment:
Love this example ❤️ 

Related, is there any way of telling ESLint that `__exeperminental*` and `__unstable*` prefixes are fine? It yells when you try to use. It is fine but a little bit annoying to use such method. At least for experimental, it could be less strict.

Reply:
> Related, is there any way of telling ESLint that `__exeperminental*` and `__unstable*` prefixes are fine? It yells when you try to use. It is fine but a little bit annoying to use such method. At least for experimental, it could be less strict.

Could you share an example? I think I've seen it before where we use the non-conventional `unstable__Whatever`, but with the correct, leading underscores, it doesn't trip ESLint's `camelcase` rule:

https://eslint.org/demo/#eyJ0ZXh0IjoiY29uc3QgeyBfX2V4cGVyaW1lbnRhbEZvbyB9ID0gd2luZG93O1xuY29uc3QgeyBleHBlcmltZW50YWxfX0ZvbyB9ID0gd2luZG93OyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6OSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJjYW1lbGNhc2UiOjJ9LCJlbnYiOnt9fX0=

Reply:
https://github.com/WordPress/gutenberg/pull/14551/files#diff-33e0ac9c44b16a617039a563d4c065f3R60

It looks like I followed the wrong usage (`unstable__bootstrapServerSideBlockDefinitions`) in the first place 🤣